packages/amplify-migration-tests/src/__tests__/update_tests/auth_migration_update.test.ts

```diff
@@ -56,43 +61,61 @@ describe('amplify auth migration', () => {
-
-    // update and push with codebase
     const authResourceName = Object.keys(meta.auth).filter(resourceName => meta.auth[resourceName].service === 'Cognito')[0];
-    // update and push with codebase
     const overridesObj: $TSAny = {
       resourceName: authResourceName,
       category: 'auth',
       service: 'cognito',
     };
-    await updateAuthWithoutCustomTrigger(projRoot, { testingWithLatestCodebase: true, overrides: overridesObj });
-    await amplifyPushAuth(projRoot, true);
+    // eslint-disable-next-line spellcheck/spell-checker
+    await updateAuthWithoutCustomTrigger(projectRoot, { testingWithLatestCodebase: true, overrides: overridesObj });
+    await amplifyPushAuth(projectRoot, true);
     const updatedFunction = await getLambdaFunction(functionName, meta.providers.awscloudformation.Region);
-    const updatedDirContents = fs.readdirSync(`${projRoot}/amplify/backend/function/${Object.keys(meta.auth)[0]}PreSignup/src`);
+    const updatedDirContents = fs.readdirSync(`${projectRoot}/amplify/backend/function/${Object.keys(meta.auth)[0]}PreSignup/src`);
     expect(updatedDirContents.includes('custom.js')).toBeFalsy();
+    // eslint-disable-next-line spellcheck/spell-checker
     expect(updatedDirContents.includes('email-filter-denylist.js')).toBeTruthy();
+    // eslint-disable-next-line spellcheck/spell-checker
     expect(updatedFunction.Configuration.Environment.Variables.MODULES).toEqual('email-filter-denylist');
   });
 
   it('...should init a project and add auth with default, and then update with latest and push', async () => {
     // add and push auth with installed cli
-    await addAuthWithDefault(projRoot, {});
-    await amplifyPushAuth(projRoot);
-    const meta = getProjectMeta(projRoot);
+    await addAuthWithDefault(projectRoot, {});
+    await amplifyPushAuth(projectRoot);
+    const meta = getProjectMeta(projectRoot);
     const authResourceName = Object.keys(meta.auth).filter(resourceName => meta.auth[resourceName].service === 'Cognito')[0];
-    // update and push with codebase
     const overridesObj: $TSAny = {
       resourceName: authResourceName,
       category: 'auth',
       service: 'cognito',
     };
-    await updateAuthWithoutTrigger(projRoot, { testingWithLatestCodebase: true, overrides: overridesObj });
-    await amplifyPushAuth(projRoot, true);
+    // eslint-disable-next-line spellcheck/spell-checker
+    await updateAuthWithoutTrigger(projectRoot, { testingWithLatestCodebase: true, overrides: overridesObj });
+    await expect(await amplifyPushAuth(projectRoot, true)).resolves.not.toThrowError();
+  });
+
+  it('...should init a project and add auth with default, and then add function and push', async () => {
+    // add and push auth with installed cli
+    await addAuthWithMaxOptions(projectRoot, {});
+    await amplifyPushAuth(projectRoot);
+    const random = Math.floor(Math.random() * 10000);
+    const fnName = `integtestfn${random}`;
```

Comment:
use uuid or nanoid instead of Math.random - otherwise we risk additional flakiness due to collisions